src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs

```diff
@@ -233,6 +224,21 @@ inlineCommonOperators = applyAll $
     go m acc (JSApp _ lhs [arg]) = go (m - 1) (arg : acc) lhs
     go _ _   _ = Nothing
 
+  inlineNonClassFunction :: (JS -> Bool) -> (JS -> JS -> JS) -> JS -> JS
+  inlineNonClassFunction p f = everywhereOnJS convert
+    where
+    convert :: JS -> JS
+    convert (JSApp _ (JSApp _ op' [x]) [y]) | p op' = f x y
+    convert other = other
+
+  isMF :: (String, String) -> JS -> Bool
```

Comment:
I was going for MF (Module and Function) and DMF (Dictionary, Module and Function), but realize that using acronyms was not very readable. I have pushed changes that better describe these functions. Thanks :)